scripts/proteinInteractionMINT/parse_mint.py

```diff
@@ -0,0 +1,418 @@
+            cur_line = get_cur_line(key, term_map[key], 'dcs:')
+            if cur_line:
+                schema_piece_list.append(cur_line)
+
+        elif key == 'interactingProtein' and term_map[key]:
+            cur_line = get_cur_line(key, term_map[key], 'dcs:bio/')
+            if cur_line:
+                schema_piece_list.append(cur_line)
+
+        elif key == 'references' and term_map[key]:
+            for cur_term in term_map[key]:
+                if cur_term:
+                    cur_line, new_reference_map = get_references(cur_term)
+                    if cur_line:
+                        schema_piece_list.append(cur_line)
+                    if new_reference_map:
+                        new_source_map[key] = new_source_map[key].update(new_reference_map)
+
+        elif key == 'identifier' and term_map[key]:
+            for cur_term in term_map[key]:
+                if cur_term:
+                    cur_line, new_identifier_map = get_identifier(cur_term)
+                    if cur_line:
+                        schema_piece_list.append(cur_line)
+                    if new_identifier_map:
+                        new_source_map[key] = new_source_map[key].update(new_identifier_map)
+
+        elif key == 'confidence' and term_map[key]:
+            item_list = []
+            for cur_term in term_map[key]:
+                if cur_term:
+                    item, new_confidence_source = get_confidence(cur_term)
+                    item_list.append(item)
+            if item_list:
+                cur_line = 'confidenceScore: ' +  ','.join(item_list)
+                schema_piece_list.append(cur_line)
+            if new_confidence_source:
+                new_source_map[key] = new_source_map[key].update(new_confidence_source)
+    return '\n'.join(schema_piece_list), new_source_map
+
+def main(argv):
+    "Main function to read the database file and generate data mcf"
+    database_file = FLAGS.database
+    psimi_to_dcid_file = FLAGS.psimi_to_dcid
+    with open(database_file, 'r') as file_object:
+        file = file_object.read()
+
+    # read the file which has paired PSI-MI and DCID.
+    # This file was generated by parse_ebi.py from EBI MI Ontology.
+    with open(psimi_to_dcid_file, 'r') as file_object:
+        psimi_to_dcid_content = file_object.read()
+
+    lines = file.split('\n')
+    psimi_to_dcid = {}
+    psimi_to_dcid_content = [line.split(': ') for line in psimi_to_dcid_content.split('\n')]
+    for line in psimi_to_dcid_content:
+        psimi_to_dcid[line[0]] = line[1]
+
+    new_source_map = {'references':{}, 'identifier':{}, 'confidence':{}}
+    mcf_list = []
+    wroing_dcid_cases = []
+    failed_cases = []
+    no_uniprot_cases = []
+    for line in lines:
+        if len(line) == 0:
+            continue
+        term = line.split('\t')
+        # check if the record has the correct UniProt Protein Name
+        if_dcid1, if_dcid2 = check_dcid(term[4]), check_dcid(term[5])
```

Comment:
Explain what the indexes 4, 5, 0, 1 mean? Or use symbolic constants.

Reply:
Added comments and examples.